lib/scheduling_middleware.rb

```diff
@@ -0,0 +1,58 @@
+# Note that this scheduling middleware only works for the FullScannerWorker
+# which contains jobs that can potentially take multiple hours to finish.
+class SchedulingMiddleware
+  def call(worker_instance, msg, queue)
+    if msg["class"] == "FullScannerWorker"
+      opts = msg["args"].last
+      # If start and end time are 0.0, no scheduling was provided. Start the
+      # job.
+      if opts["utc_start_test"] == 0.0 && opts["utc_end_test"] == 0.0
+        return yield
+      end
+
+      utc_start_time = decimal_to_utc_time(opts["utc_start_test"])
+      utc_end_time = decimal_to_utc_time(opts["utc_end_test"])
+
+      time_now = Time.now.utc
+
+      #TODO: don't forget about minutes
+      if (0..utc_start_time.hour-1).cover? time_now.hour
+        utc_start_time -= 1.day
+      else
+        # If the end time is less than the start time, add a day as this was
+        # meant to be a cross-day timeframe
+        utc_end_time += 1.day if utc_end_time < utc_start_time
```

Comment:
Feel like adding <= to line 24 and removing lines 9-11 would make it less obvious that 0.0 start and end values means that no scheduling was provided. 9-11. 19-25 can be kept as strict overlap checking logic and 9-11 as no scheduling logic.